control-plane/pkg/reconciler/broker/broker.go

```diff
@@ -226,7 +235,34 @@ func (r *Reconciler) reconcileKind(ctx context.Context, broker *eventing.Broker)
 	}
 
 	ingressHost := network.GetServiceHostname(r.Env.IngressName, r.DataPlaneNamespace)
-	address := receiver.Address(ingressHost, broker)
+
+	transportEncryptionFlags := feature.FromContext(ctx)
+	if transportEncryptionFlags.IsPermissiveTransportEncryption() {
+		caCerts, err := r.getCaCerts()
+		if err != nil {
+			return err
+		}
+
+		httpAddress := receiver.HTTPAddress(ingressHost, broker)
+		httpsAddress := receiver.HTTPSAddress(ingressHost, broker, caCerts)
+		broker.Status.Address = &httpAddress
+		broker.Status.Addresses = []pkgduckv1.Addressable{httpAddress, httpsAddress}
+	} else if transportEncryptionFlags.IsStrictTransportEncryption() {
+		caCerts, err := r.getCaCerts()
+		if err != nil {
+			return err
+		}
+
+		httpsAddress := receiver.HTTPSAddress(ingressHost, broker, caCerts)
+		broker.Status.Address = &httpsAddress
+		broker.Status.Addresses = []pkgduckv1.Addressable{httpsAddress}
+	} else {
+		httpAddress := receiver.HTTPAddress(ingressHost, broker)
+		broker.Status.Address = &httpAddress
+		broker.Status.Addresses = []pkgduckv1.Addressable{httpAddress}
+	}
```

Comment:
We need to set the address only after the prober is successful

Reply:
Good point, I've updated the order of events so that it makes more sense